spot-contracts/contracts/PerpetualTranche.sol

```diff
@@ -333,14 +402,25 @@ contract PerpetualTranche is
     /// @notice Updates the reference to the keeper.
     /// @param newKeeper The address of the new keeper.
     function updateKeeper(address newKeeper) public virtual onlyOwner {
-        if (newKeeper == address(0)) {
-            revert UnacceptableReference();
-        }
         address prevKeeper = keeper;
         keeper = newKeeper;
         emit UpdatedKeeper(prevKeeper, newKeeper);
     }
 
+    /// @notice Updates the authorized roller set.
+    /// @dev CAUTION: If the authorized roller set is empty, all rollers are authorized.
+    /// @param roller The address of the roller.
+    /// @param authorize If the roller is to be authorized or unauthorized.
+    function authorizeRoller(address roller, bool authorize) public onlyOwner {
+        if (authorize && !_rollers.contains(roller)) {
+            _rollers.add(roller);
+        } else if (!authorize && _rollers.contains(roller)) {
+            _rollers.remove(roller);
+        }
```

Comment:
We could add an `else return` so we only issue an event when something changes.

Reply:
done